ipv8/attestation/trustchain/block.py

```diff
@@ -44,19 +44,18 @@ def __init__(self, data=None, serializer=Serializer()):
             # debug stuff
             self.insert_time = None
         else:
-            _, self.transaction = decode(str(data[1]))
+            _, self.transaction = decode(data[1] if isinstance(data[1], bytes) else str(data[1]))
             (self.type, self.public_key, self.sequence_number, self.link_public_key, self.link_sequence_number,
              self.previous_hash, self.signature, self.timestamp, self.insert_time) = (data[0], data[2], data[3],
                                                                                       data[4], data[5], data[6],
                                                                                       data[7], data[8], data[9])
-            if isinstance(self.public_key, database_blob):
-                self.public_key = str(self.public_key)
-            if isinstance(self.link_public_key, database_blob):
-                self.link_public_key = str(self.link_public_key)
-            if isinstance(self.previous_hash, database_blob):
-                self.previous_hash = str(self.previous_hash)
-            if isinstance(self.signature, database_blob):
-                self.signature = str(self.signature)
+            self.type = self.type if isinstance(self.type, bytes) else str(self.type)
```

Comment:
RIP to the elegance of TrustChain code :(

Reply:
By all means, look at the code one line up.